### PR TITLE
Task/psgs 63 jsons

### PR DESCRIPTION
**Pull request by maciejbak85**

### CODE REVIEW

File: src/main/scala/iog/psg/cardano/util/Trace.scala [hunk 1]
```diff
@@ -33,23 +38,23 @@ trait Trace extends AutoCloseable {
...
-  override def apply(s: Object): Unit = ()
+  override def apply(s: String): Unit = ()
   override def close(): Unit = ()
 
   override def withTrace(other: Trace): Trace = other
 }
 
 object ConsoleTrace extends Trace {
-  override def apply(s: Object): Unit = println(s)
+  override def apply(s: String): Unit = println(s: String)
```
Comment (maciejbak85) [inline src/main/scala/iog/psg/cardano/util/Trace.scala:48]: this s: String can be removed now, leftover will push change in a sec

File: src/main/scala/iog/psg/cardano/util/Trace.scala [hunk 2]
```diff
@@ -2,19 +2,24 @@ package iog.psg.cardano.util
...
+  implicit def s2Str[A](s: A)(implicit enc: Encoder[A]): String = s.asJson.spaces2
+  def apply(s: String): Unit
 
   def withTrace(other: Trace): Trace = other match {
     case NoOpTrace => this
     case _ =>
       new Trace {
-        override def apply(s: Object): Unit = {
+
+        override def apply(s: String): Unit = {
```
Comment (mcsherrylabs) [inline src/main/scala/iog/psg/cardano/util/Trace.scala:22]: FYI, this was a type 'Object' specifically to handle the case classes.

**Review by mcsherrylabs — CHANGES_REQUESTED** [review]
Fix the warning! That's all.

File: src/main/scala/iog/psg/cardano/util/Trace.scala [hunk 3]
```diff
@@ -2,19 +2,24 @@ package iog.psg.cardano.util
...
+
 import scala.util.Try
+import io.circe.syntax._
 
 trait Trace extends AutoCloseable {
 
   parent =>
 
-  def apply(s: Object): Unit
+  implicit def s2Str[A](s: A)(implicit enc: Encoder[A]): String = s.asJson.spaces2
```
[the remark this answers is not in the document]
Comment (maciejbak85) [reply src/main/scala/iog/psg/cardano/util/Trace.scala:14]: hmm for some reason I have this enabled and I dont see it, we can enable them but well I am fine with implicit class, at least it wont be done "magically"

**Review by mcsherrylabs — APPROVED** [review]
Horray! Please merge!

**Pull request merged**